src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,114 @@
+package duke;
+
+import duke.commands.AddTaskCommand;
+import duke.commands.ByeCommand;
+import duke.commands.Command;
+import duke.commands.DeleteCommand;
+import duke.commands.DoneCommand;
+import duke.commands.FindTasksCommand;
+import duke.commands.ListCommand;
+
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+
+/**
+ * This class represents a Parser used to parse user input and return Commands that can be executed by Duke.
+ */
+public class Parser {
+    /**
+     * Parses user input into a Command that should be executed by the Duke chatbot.
+     * 
+     * @param answer User input parsed into a String.
+     * @return Command that is to be executed by Duke based on user input.
+     * @throws DukeException
+     */
+    public static Command parse(String answer) throws DukeException {
+        String[] parts = answer.split(" ");
+        String command = parts[0];
+        String taskDetails = "";
+        if (answer.contains(" ")) {
+            taskDetails = answer.substring(answer.indexOf(" ") + 1);
+        }
+        switch (command) {

```

Comment:
I think you can remove the indentation for case and default of switch?
![image](https://user-images.githubusercontent.com/54053213/131605144-88758d3d-4014-4133-b14d-7f88abb7a752.png)